### PR TITLE
Migrate async/iteration to non-null by default

### PR DESCRIPTION
**Pull request by cbracken**
Partial fix towards #606.

### CODE REVIEW

File: lib/src/async/iteration.dart [hunk 1]
```diff
@@ -42,7 +40,7 @@ Future<S> _reduceAsync<S, T>(
 
 /// Schedules calls to [action] for each element in [iterable]. No more than
 /// [maxTasks] calls to [action] will be pending at once.
-Future<Null> forEachAsync<T>(Iterable<T> iterable, AsyncAction<Null, T> action,
```
Comment (cbracken) [inline lib/src/async/iteration.dart:45]: I wonder if we should just deprecate this in Quiver 2.0 and delete it here? Future.forEach](https://api.flutter.dev/flutter/dart-async/Future/forEach.html) is pretty much the same thing though it doesn't have the task limit, which I suppose could be useful for limiting things that involve system resources like file handles or sockets.

Comment (yjbanov) [reply lib/src/async/iteration.dart:45]: Do you mean 3.0? I'm OK with deprecating. I think `package:pool` is the de facto standard for this sort of thing.

**Review by yjbanov — APPROVED** [review]
lgtm

**Comment by cbracken** [comment]
> Do you mean 3.0?

I mean deprecate on the main branch, release a new release, then delete here.

**Pull request merged**